### PR TITLE
Don't run `apt update` if it's been run in the past hour

### PR DESCRIPTION
**Pull request by Efreak**
Don't run `apt update` if it's been run in the past hour.
Also added a separate update command to force updating anyways.

Reasoning: updating the cache can take a while even if you're there's no changes, due to multiple repos. Sometimes I spend quite a while installing packages one at a time so I can install something via pip/other tools. This should speed that up by using the cache for longer sessions.

An alternative might be to export a variable and check for that, as presumably most sessions don't last long enough for package management to be affected.

### CODE REVIEW

**Comment by tomty89** [comment]
No offense, it doesn't exactly make sense (the "time-based" part).

With that said, the fact that `pkg` does `apt update` for searching and installing is kind of a bad idea. It's convenient, but it's bad.

I would recommend stop doing so leave up* = update + upgrade. (I don't think we need upd* = update for `pkg` as it's kind of a bad practice in Termux, as it's like `pacman -Sy` in Arch, which is also one of the reasons that `pkg` should do `apt update` for searching and installing.)

**Comment by buttaface** [comment]
I don't use pkg because of its annoying network hit on every run, only use apt instead.

File: packages/termux-tools/pkg [hunk 1]
```diff
@@ -26,21 +27,28 @@ assert_not_root() {
   fi
 }
 
+check_pkgcache(){
+  if [ -z "$(find $HOME/../usr/var/cache/apt/pkgcache.bin -mmin -60)" ]; then
```
Comment (xeffyr) [inline packages/termux-tools/pkg]: Replace `$HOME/../usr` with `$PREFIX`.

**Comment by xeffyr** [comment]
If you don't like `pkg` for whatever reason, you can still use `apt`.

The purpose of `pkg` is to solve the following issues:

* `apt` should not be run as root
* Make sure that package lists always available. Doing `apt search` after fresh Termux installation you will see nothing as well as you won't be able to install anything without running `apt update` first.
* Make sure that package lists always up-to-date. **We don't keep old versions !** Trying to install something may result in 404 errors which confuses some users.
* Simplify command line to use package manager. Instead of `apt update && apt install ....` you need to run just `pkg install`.

**Comment by buttaface** [comment]
That's all well and good for noobs, but I guess as a more advanced user, I have no problem using apt directly, though I sometimes have to drop down to dpkg too. :wink:

**Comment by tomty89** [comment]
> Trying to install something may result in 404 errors which confuses some users.

Yet in Arch we still strongly oppose running `pacman -Sy` under normal circumstances, which we have a good reason to. It's simply how rolling releases work: both (update + upgrade) or nothing.

**Comment by Efreak** [comment]
If one hour is too long, I'm equally happy with half that, or even less. 

> apt should not be run as root (well, I can "lock" apt from doing this in its source, but there will be issues about unusable package manager under ADB).

This is why I use pkg and not apt. I use `su` quite frequently.

> Make sure that package lists always available. Doing apt search after fresh Termux installation you will see nothing as well as you won't be able to install anything without running apt update first.

Shouldn't there not be a package listing if you've just installed termux and haven't run the update yet? If there's no package cache, it should update given my proposal. If the package cache exists, then if you've got another file that's added by termux at installation and rarely/never changed, we can check the time on that as well. Will update requested fix when I have a chance.

File: packages/termux-tools/pkg (hunk 1 above)
Comment (Efreak) [reply packages/termux-tools/pkg]: done. it looks like I force-pushed it for some reason, can't remember why.

**Pull request merged**